webtau-groovy/src/main/groovy/org/testingisdocumenting/webtau/cli/repl/ReplScenariosSelector.groovy

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2020 webtau maintainers
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.testingisdocumenting.webtau.cli.repl
+
+
+import org.testingisdocumenting.webtau.runner.standalone.StandaloneTest
+
+import java.util.regex.Pattern
+
+class ReplScenariosSelector {
+    private final List<StandaloneTest> availableScenarios
+    private final List<SelectInputAndResult> inputAndResults
+
+    ReplScenariosSelector(List<StandaloneTest> availableScenarios, Object... selectors) {
+        this.availableScenarios = availableScenarios
+        this.inputAndResults = selectors.collect { selectScenario(it) }.flatten()
+    }
+
+    List<SelectInputAndResult> getInputAndResults() {
+        return inputAndResults
+    }
+
+    private List<SelectInputAndResult> selectScenario(Object selector) {
+        if (selector instanceof String) {
+            return selectSingleScenarioByRegexp(selector)
+        }
+
+        if (selector instanceof Integer) {
+            return selectSingleScenarioByIdx(selector)
+        }
+
+        if (selector instanceof Map) {
+            return selectSingleScenariosByRange(selector)
+        }
+
+        return [new SelectInputAndResult(selector, -1, null)]
+    }
+    
+    private List<SelectInputAndResult> selectSingleScenarioByRegexp(String regexp) {
+        def pattern = Pattern.compile(regexp, Pattern.CASE_INSENSITIVE)
+        def idx = availableScenarios.findIndexOf { pattern.matcher(it.scenario) }
+        def found = idx == -1 ? null : availableScenarios[idx]
+
+        return [new SelectInputAndResult(regexp, idx, found)]
+    }
+
+    private List<SelectInputAndResult> selectSingleScenarioByIdx(int idx) {
+        def isInRange = idx >= 0 && idx < availableScenarios.size()
+        def found = isInRange ? availableScenarios[idx] : null
+
+        return [new SelectInputAndResult(idx, found ? idx : -1, found)]
+    }
+
+    private List<SelectInputAndResult> selectSingleScenariosByRange(Map range) {
+        if (range.size() != 1) {
+            return [new SelectInputAndResult(range, -1, null)]
+        }
+
+        def entrySet = range.iterator().next()
+        return selectSingleScenariosByRange(entrySet.key, entrySet.value)
+    }
+
+    private List<SelectInputAndResult> selectSingleScenariosByRange(Object from, Object to) {
```

Comment:
This one actually returns more than one thing so it's not "single".

Reply:
Thanks for catching. Yes this is not single. The other ones return lists so it is easier to combine in one go and flatten

Reply:
OK, it's a bit weird to return a list which always has one thing.  Looking at the usage we just call these things from a big if/else if/else where it would be easy to just warp the single element in a list.  I would personally find that less confusing (also if we have tests for this class directly, you won't have to test the list is a single item list :) ).  That said, it's not a huge deal so I won't block the PR on it.